sklearn/neighbors/base.py

```diff
@@ -776,16 +776,33 @@ def fit(self, X, y):
         else:
             self.outputs_2d_ = True
 
-        check_classification_targets(y)
+        try:
+            check_classification_targets(y)
+        except ValueError as e:
```

Comment:
I find this part of the code ugĺy. Even more now that we added `'sparse-muliclass-multioutput'` as a `type_of_target(y)` [here](https://github.com/scikit-learn/scikit-learn/pull/9059/files#diff-c569d578c4c466c2b310da6a2b06e1caR252). Wouldn't it be a way to make use of the message from type_of_target to avoid checking again `issparse(y) and self.outputs_2d_`  ? 
cc: @jnothman, @ogrisel